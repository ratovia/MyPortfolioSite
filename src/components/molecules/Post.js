import React from "react"
import styled from "styled-components"
import { Photograph } from "../atoms/Photograph"
import { TypographyH5, TypographyH6 } from "../atoms/Typography"

const PostArea = styled.div`
  width: 100%;
  height: 470px;
  position: relative;
`

const PostDate = styled.div`
  position: absolute;
  top: 40px;
  left: 40px;
`

const PostTitle = styled.div`
  position: absolute;
  bottom: 40px;
  right: 40px;
`

export const Post = ({ imageUrl, date, title }) => {
  return (
    <PostArea>
      <Photograph imageUrl={imageUrl}></Photograph>
      <PostDate>
        <TypographyH6 label={date} color="#CBCBCB"></TypographyH6>
      </PostDate>
      <PostTitle>
        <TypographyH5 label={title} color="white"></TypographyH5>
      </PostTitle>
    </PostArea>
  )
}
