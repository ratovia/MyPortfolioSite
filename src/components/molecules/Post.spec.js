import React from "react"
import renderer from "react-test-renderer"
import { Post } from "./Post"
import backgroundImg from "../../stories/assets/bg.jpg"

describe("Post", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Post imageUrl={backgroundImg} date="2021-1-1" title="Hello World" />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
