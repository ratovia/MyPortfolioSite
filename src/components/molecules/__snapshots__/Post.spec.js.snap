// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders correctly 1`] = `
<div
  className="sc-jRQAMF kNwGDD"
>
  <img
    alt="アイコン"
    height="100%"
    src="test-file-stub"
    width="100%"
  />
  <div
    className="sc-gKckTs elISde"
  >
    <h6
      className="sc-eCImvq kPbozt"
    >
      <p
        className="sc-gsDJrp dYWiJh"
        color="#CBCBCB"
      >
        2021-1-1
      </p>
    </h6>
  </div>
  <div
    className="sc-iCfLBT gwOeYc"
  >
    <h5
      className="sc-hKwCoD gWiDwP"
    >
      <p
        className="sc-gsDJrp jWZtA"
        color="white"
      >
        Hello World
      </p>
    </h5>
  </div>
</div>
`;
